### PR TITLE
Add blake3 python binding

### PR DESCRIPTION
**Pull request by maartenbreddels**
@oconnor663 I added you as maintainer (basically you have permission to do a new release, but if you don't want to I can remove you).

### CODE REVIEW

**Comment by oconnor663** [comment]
Thanks! Should there be explicit dependencies on the Rust tools necessary to build? Or do we assume that we're able to pull a binary wheel from PyPI?

**Comment by maartenbreddels** [comment]
Yes, the failure seems to be a recipe error, but I added rust as a build dependency now, conda-forge always builds from source.

File: recipes/blake3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - maturin
+    - pip
+    - python
+    - rust
```
Comment (scopatz) [inline recipes/blake3/meta.yaml:22]: Can this be noarch:python? Or no because of the rust?

**Comment by maartenbreddels** [comment]
@oconnor663 does it make sense for this package to be named 'blake3', or should it have py in the name, e.g. blake3-py ? Would it make sense to have a blake3 package for instance that contains the b3sum cmd line util? Or maybe different language bindings. For pypi it makes sense, but conda-forge is not Python specific.

File: recipes/blake3/meta.yaml (hunk 1 above)
Comment (oconnor663) [reply recipes/blake3/meta.yaml:24]: Right, this library is bindings for the underlying Rust code, so the build artifacts are arch-specific.

**Comment by oconnor663** [comment]
I don't know anything about conda-forge's conventions, so I'll defer to you.

File: recipes/blake3/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/blake3/meta.yaml:24]: Got it, cool

File: recipes/blake3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 99d688bd1b673de332a2927665e0e82eefae79b18b7c03564c5907d716e473b8
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
```
Comment (xhochy) [inline recipes/blake3/meta.yaml:18]: Nowadays there is a compiler macro for rust.
```suggestion
  build:
   - {{ compiler('rust') }}
```

**Comment by maartenbreddels** [comment]
@xhochy this dropped off my radar, could you reopen?

**Comment by xhochy** [comment]
No, I cannot 

image

**Comment by maartenbreddels** [comment]
that's true, I guess I have to reopen.

**Comment by maartenbreddels** [comment]
Continued in #14985

**Pull request closed without merging**